### PR TITLE
volume should only be removed from a running sandbox

### PR DESCRIPTION
**Pull request by Crazykev**
Signed-off-by: Crazykev <crazykev@zju.edu.cn>

### CODE REVIEW

File: daemon/pod/decommission.go [hunk 1]
```diff
@@ -280,31 +296,6 @@ func (p *XPod) RemoveContainer(id string) error {
...
-			if !removed {
-				v.Log(DEBUG, "volume did not unplug because it is in use")
-				continue
-			}
-			v.Log(DEBUG, "volume unplugged")
-			removedVols = append(removedVols, cv.Name)
-		}
-	}
-
-	//remove volumes from daemondb
```
Comment (feiskyer) [inline daemon/pod/decommission.go]: Those are removed accidently or intended?

Comment (Crazykev) [inline daemon/pod/decommission.go]: @feiskyer intendedly, I just notice remove container shouldn't remove related volume from pod(at least there is no such logic here), so this should not store to db.

Comment (feiskyer) [inline daemon/pod/decommission.go]: >  remove container shouldn't remove related volume from pod

Why? I think if a volume is not referenced by any containers, it should be removed.

Comment (Crazykev) [inline daemon/pod/decommission.go]: I'm not sure what the original design of this, could later created container use those volumes meant to be removed?  If not, yeah, we should remove it from pod and db.

**Comment by Crazykev** [comment]
@feiskyer I've reverted changes about removing volumes from db, if it's needed, will open another PR to fix.
also /cc @gnawux do you have any comment on the disscusion?

**Comment by gnawux** [comment]
Thanks @Crazykev , I don't have any more comment on the code. 

If not too complicated, could you help adding the case in #537 to the test cases of hyperd?

**Comment by Crazykev** [comment]
I have no idea of why `not a directory` happens, while this one I can do something

**Comment by feiskyer** [comment]
LGTM. Let's merge this first and check whether there are other potential problems by adding loads to hyperd (e.g. via kubernetes tests).

**Pull request merged**